lib/resty/redis.lua

```diff
@@ -82,11 +86,47 @@ function _M.connect(self, ...)
         return nil, "not initialized"
     end
 
+    local params = {...}
+
+    if not params or #params < 2 then
+        return nil, "invalid arguments"
+    end
+
     self._subscribed = false
 
-    return sock:connect(...)
-end
+    local host = params[1]
+    local connect_opts, opts, err
+    if startswith(host, "unix:") then
+        opts = params[2] or {}
+        connect_opts = {
+            pool = opts.pool,
+            poolsize = opts.poolsize,
+            backlog = opts.backlog
+        }
+        ok, err = sock:connect(host, connect_opts)
+    else
+        local port = params[2] or 6379
+        opts = params[3] or {}
+        connect_opts = {
+            pool = opts.pool,
+            poolsize = opts.poolsize,
+            backlog = opts.backlog
+        }
+        ok, err = sock:connect(host, port, connect_opts)
```

Comment:
@vinayakhulawale After using `sock:connect` at this `if` block, you need to check if that function return `ok` or `err` before go to ssl handler. Maybe it can be check after `end` command, at line 117.

Reply:
thanks, added error handling